### DISCUSSION TITLE
basic question regarding fltk-webview on linux

### DISCUSSION THREAD

**Discussion by interstellarvoyager** [fltk-rs/fltk-rs#1269] [Q&A]
how do i "pass the GDK_BACKEND=x11 environment variable"  ?

**Comment by MoAlyousef** [comment]
It depends if your desktop environment is wayland or not, since the embedding doesn't work natively on wayland, and will require a change in the GDK_BACKEND of the gtkwebkit. On a X11 system, you can ignore all the following.

You can set the env variable globally but this will likely affect all apps:
```
export GDK_BACKEND=x11
```
Otherwise you can set it only for your app:
So assuming your app's binary is called `myapp`
```
GDK_BACKEND=x11 myapp # if it's globally installed
# or
GDK_BACKEND=x11 ./myapp
```
If you're running from cargo as during development:
```
GDK_BACKEND=x11 cargo run
```
If you're deploying the app as some deployed format (like appimage for example), you can set the environment variable in the manifest or wrap the app in a script which calls your binary preceded with  `GDK_BACKEND=x11`, and make the script the entry for your app.
